core/Azure.Mcp.Core/src/Services/Azure/Tenant/TenantServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Mcp.Core.Services.Azure.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Azure.Mcp.Core.Services.Azure.Tenant;
+
+/// <summary>
+/// Extension methods for configuring Azure tenant services.
+/// </summary>
+public static class TenantServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds <see cref="TenantService"/> as <see cref="ITenantService"/> with lifetime
+    /// <see cref="ServiceLifetime.Singleton"/> into the service collection.
+    /// </summary>
+    /// <param name="services">The service collection.</param>
+    /// <returns>The service collection.</returns>
+    /// <remarks>
+    /// <para>
+    /// This method follows the dependency graph pattern by ensuring all dependencies of
+    /// <see cref="TenantService"/> are registered by calling their respective extension methods.
+    /// </para>
+    /// <para>
+    /// Dependencies registered:
+    /// </para>
+    /// <list type="bullet">
+    /// <item>
+    /// <description><see cref="IAzureTokenCredentialProvider"/> via <see cref="Authentication.AuthenticationServiceCollectionExtensions.AddSingleIdentityTokenCredentialProvider"/>.
+    /// This can be overridden using <see cref="Authentication.AuthenticationServiceCollectionExtensions.AddAzureTokenCredentialProvider"/>
+    /// based on parsed command line arguments and environment variables.
+    /// </description>
+    /// </item>
+    /// </list>
+    /// </remarks>
+    public static IServiceCollection AddAzureTenantService(this IServiceCollection services)
+    {
+        // !!! HACK !!!
+        // Program.cs for the CLI servers have their own DI containers vs ServiceStartCommand.
```

Comment:
Completely out of scope for this work, but for future GA releases, we might want to revisit whether we still need to support the fully functional CLI mode (beyond `server start` and `--version`). I recall GHCP4z used to rely on the CLI mode at one point, but that’s no longer the case. Feel free to resolve the comment - I just wanted to highlight the additional overhead of maintaining a mode (CLI) whose value for MCP server users is unclear.

Reply:
I agree, but I'm very biased with my only focus being on `server start` as a scenario. @joshfree mentioning you because this focusing on the azmcp.exe being exclusively an MCP server would be very nice IMO but you have a clearer product vision.